### PR TITLE
Monodepth2 mono

### PR DESCRIPTION
**Pull request by KuangHaofei**
This PR provide the training and test pipeline of monodepth2 with monocular mode and monocular+stereo mode.
Including:
  - 4 new models: 2 depth models; 2 pose models
  - calculate loss by using pose net
  - pose net test code

### CODE REVIEW

File: gluoncv/model_zoo/model_zoo.py [hunk 1]
```diff
@@ -369,7 +369,14 @@
     'hrnet_w18_small_v2_s' : hrnet_w18_small_v2_s,
     'hrnet_w48_s' : hrnet_w48_s,
     'siamrpn_alexnet_v2_otb15': siamrpn_alexnet_v2_otb15,
-    'monodepth2_resnet18_kitti_stereo_640x192': get_monodepth2_resnet18_kitti_stereo_640x192
+    'monodepth2_resnet18_kitti_stereo_640x192': get_monodepth2_resnet18_kitti_stereo_640x192,
+    'monodepth2_resnet18_kitti_mono_640x192': get_monodepth2_resnet18_kitti_mono_640x192,
+    'monodepth2_resnet18_kitti_mono+stereo_640x192':
+        get_monodepth2_resnet18_kitti_mono_stereo_640x192,
+    "monodepth2_resnet18_posenet_kitti_mono_640x192":
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/model_zoo.py:376]: Can you make double quote to single quote for consistency?

File: gluoncv/model_zoo/monodepthv2/monodepth2.py [hunk 2]
```diff
@@ -105,12 +105,13 @@ def get_monodepth2(backbone='resnet18', pretrained_base=True, num_input_images=1
     """
     acronyms = {
         'kitti_stereo_640x192': 'kitti_stereo_640x192',
+        'kitti_mono_640x192': 'kitti_mono_640x192',
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/monodepthv2/monodepth2.py:108]: I didn't see any difference between the key and value here. I think they are the same string. May I ask why do we need this `acronyms ` dictionary?

File: gluoncv/model_zoo/model_zoo.py (hunk 1 above)
Comment (KuangHaofei) [reply gluoncv/model_zoo/model_zoo.py]: Modified, thanks for your reminder!

File: gluoncv/model_zoo/monodepthv2/monodepth2.py (hunk 2 above)
Comment (KuangHaofei) [reply gluoncv/model_zoo/monodepthv2/monodepth2.py]: I refer to the code of segmentation model zoo here, but acronyms are not needed actually. Modified, thanks for reminding!

**Comment by mli** [comment]
Job PR-1428-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1428/5/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**